tools/cli/adminDBTimers.go

```diff
@@ -0,0 +1,140 @@
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package cli
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/gocql/gocql"
+	"github.com/pkg/errors"
+	"github.com/urfave/cli"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/uber/cadence/common/log/loggerimpl"
+	"github.com/uber/cadence/common/persistence"
+	cassp "github.com/uber/cadence/common/persistence/cassandra"
+)
+
+// AdminTimers is used to list scheduled timers.
+func AdminTimers(c *cli.Context) {
+	lowerShardBound := c.Int(FlagLowerShardBound)
+	upperShardBound := c.Int(FlagUpperShardBound)
+	concurrency := c.Int(FlagConcurrency)
+	batchSize := c.Int(FlagBatchSize)
+	startDate := c.String(FlagStartDate)
+	endDate := c.String(FlagEndDate)
+	domainID := c.String(FlagDomainID)
+	skipErrMode := c.Bool(FlagSkipErrorMode)
+
+	ctx, cancel := newContextForLongPoll(c)
+	defer cancel()
+
+	st, err := parseSingleTs(startDate)
+	if err != nil {
+		ErrorAndExit("wrong date format for "+FlagStartDate, err)
+	}
+	et, err := parseSingleTs(endDate)
+	if err != nil {
+		ErrorAndExit("wrong date format for "+FlagEndDate, err)
+	}
+
+	session := connectToCassandra(c)
+	defer session.Close()
+
+	group, ctx := errgroup.WithContext(ctx)
+	throttle := make(chan struct{}, concurrency)
+	for shardID := lowerShardBound; shardID < upperShardBound; shardID++ {
+		shardID := shardID
+		group.Go(func() error {
+			throttle <- struct{}{}
+			defer func() {
+				<-throttle
+			}()
+
+			return printTimers(session, shardID, batchSize, st, et, domainID, skipErrMode)
+		})
+	}
+
+	if err := group.Wait(); err != nil {
+		ErrorAndExit("getting timers", err)
+	}
+}
+
+func printTimers(
+	session *gocql.Session,
+	shardID, batchSize int,
+	startDate, endDate time.Time,
+	domainId string,
+	skipErrMode bool,
+) error {
+	execStore, err := cassp.NewWorkflowExecutionPersistence(shardID, session, loggerimpl.NewNopLogger())
+
+	if err != nil {
+		err = errors.Wrapf(err, "cannot get execution manager for shardId %d", shardID)
+		if !skipErrMode {
+			return err
+		}
+		fmt.Println(err.Error())
+		return nil
+	}
+
+	var token []byte
+	isFirstIteration := true
+	for isFirstIteration || len(token) != 0 {
+		isFirstIteration = false
+		req := persistence.GetTimerIndexTasksRequest{
+			MinTimestamp:  startDate,
+			MaxTimestamp:  endDate,
+			BatchSize:     batchSize,
+			NextPageToken: token,
+		}
+		resp, err := execStore.GetTimerIndexTasks(&req)
+
+		if err != nil {
+			return errors.Wrapf(err, "cannot get timer tasks for shardId %d", shardID)
+		}
+
+		token = resp.NextPageToken
+
+		for _, t := range resp.Timers {
+			if len(domainId) > 0 && t.DomainID != domainId {
+				continue
+			}
+
+			data, err := json.Marshal(t)
+			if err != nil {
+				err = errors.Wrap(err, "cannot marshal timer to json")
+				if !skipErrMode {
+					return err
+				}
+				fmt.Println(err.Error())
+			} else {
+				fmt.Println(string(data))
```

Comment:
This looks like it just prints timers data? What if I want output like "give me the RPS for timers for the next day grouped by hour"?

Do users need to do some fancy grep and awk in order to answer these queries?

Reply:
I assume there will be part 2 that includes this?